terraform/ecs.tf

```diff
@@ -0,0 +1,182 @@
+resource "aws_ecr_repository" "main" {
+  name                 = "${var.name}-${var.environment}"
+  image_tag_mutability = "MUTABLE"
+}
+
+resource "aws_ecr_lifecycle_policy" "main" {
+  repository = aws_ecr_repository.main.name
+
+  policy = jsonencode({
+    rules = [{
+      rulePriority = 1
+      description  = "keep last 10 images"
+      action = {
+        type = "expire"
+      }
+      selection = {
+        tagStatus   = "any"
+        countType   = "imageCountMoreThan"
+        countNumber = 10
+      }
+    }]
+  })
+}
+
+resource "aws_ecs_cluster" "main" {
+  name = "${var.name}-cluster-${var.environment}"
+}
+
+resource "aws_cloudwatch_log_group" "ecs_log_group" {
+  name = "${var.name}-ecs-logs-${var.environment}"
+  tags = {
+    Name = "${var.name}-ecs-logs-${var.environment}"
+  }
+}
+resource "aws_ecs_task_definition" "main" {
+  network_mode             = "awsvpc"
+  family                   = "terraformlearning"
+  requires_compatibilities = ["FARGATE"]
+  cpu                      = 256
+  memory                   = 512
+  execution_role_arn       = aws_iam_role.ecs_task_execution_role.arn
+  task_role_arn            = aws_iam_role.ecs_task_role.arn
+  container_definitions = jsonencode([{
+    name      = "${var.name}-container-${var.environment}"
+    image     = "${var.container_image}:latest"
+    essential = true
+    //environment = var.environment
+    portMappings = [{
+      protocol      = "tcp"
+      containerPort = var.container_port
+      hostPort      = var.container_port
+    }]
+    logConfiguration = {
+      logDriver = "awslogs"
+      options = {
+        awslogs-group         = "${var.name}-ecs-logs-${var.environment}"
+        awslogs-region        = "us-east-2",
+        awslogs-stream-prefix = "ecs"
+      }
+    }
+  }])
+}
+
+resource "aws_iam_role" "ecs_task_role" {
+  name = "${var.name}-ecsTaskRole"
+
+  assume_role_policy = <<EOF
```

Comment:
It doesn't really matter much, but you can use the jsonencode() function here as well. 👍🏻